packages/vscode-extension/src/project/project.ts

```diff
@@ -585,17 +585,22 @@ export class Project
     }
   };
 
-  private checkIfNativeChanged = throttle(async () => {
-    if (!this.detectedFingerprintChange && this.projectState.selectedDevice) {
-      if (await didFingerprintChange(this.projectState.selectedDevice.platform)) {
-        this.detectedFingerprintChange = true;
+  private checkIfNativeChanged = throttle(async (cancelToken: CancelToken) => {
+    if (!this.buildCacheInvalidated && this.projectState.selectedDevice) {
+      // TODO: should be refactored to not create a new PlatformBuildCache
+      // instance every time
+      const platform = this.projectState.selectedDevice.platform;
+      const buildCache = new PlatformBuildCache(platform, cancelToken);
```

Comment:
responding to your comment here: maybe we could have a static method in `PlatformBuildCache.get(platform)` that returns an instance. This way we can hide the fact it is a stateless object and avoid constructing it in multiple places

Reply:
This works quite nicely but we now split getting fingerprint and storing the build (which is correct) so the value of the class is reduced. I think it's still marginally better than standalone functions but not by much.